boot/bootutil/include/bootutil/image.h

```diff
@@ -149,8 +149,8 @@ struct image_tlv {
     uint16_t it_len;    /* Data length (not including TLV header). */
 };
 
-#define IS_ENCRYPTED(hdr) (((hdr)->ih_flags && IMAGE_F_ENCRYPTED_AES128) \
-                        || ((hdr)->ih_flags && IMAGE_F_ENCRYPTED_AES256))
+#define IS_ENCRYPTED(hdr) (((hdr)->ih_flags & IMAGE_F_ENCRYPTED_AES128) \
```

Comment:
This is just a bug, and should probably be a separate commit that just fixes it (along with an explanation).

Reply:
Just created https://github.com/mcu-tools/mcuboot/pull/1072 to fix this